<?xml version="1.0" encoding="UTF-8"?>
<ui version="4.0">
 <class>Window</class>
 <widget class="QWidget" name="Window">
  <property name="geometry">
   <rect>
    <x>0</x>
    <y>0</y>
    <width>1028</width>
    <height>899</height>
   </rect>
  </property>
  <property name="windowTitle">
   <string>Window</string>
  </property>
  <layout class="QGridLayout" name="gridLayout">
   <item row="0" column="0">
    <layout class="QHBoxLayout" name="horizontalLayout">
     <item>
      <widget class="QTreeView" name="treeView">
       <property name="minimumSize">
        <size>
         <width>400</width>
         <height>480</height>
        </size>
       </property>
      </widget>
     </item>
     <item>
      <widget class="MyGLWidget" name="myGLWidget" native="true">
       <property name="minimumSize">
        <size>
         <width>480</width>
         <height>480</height>
        </size>
       </property>
      </widget>
     </item>
    </layout>
   </item>
   <item row="1" column="0" alignment="Qt::AlignTop">
    <widget class="QGroupBox" name="TransformsBox">
     <property name="title">
      <string>Transformations</string>
     </property>
     <layout class="QGridLayout" name="gridLayout_3">
      <item row="5" column="1">
       <widget class="QWidget" name="display" native="true">
        <property name="minimumSize">
         <size>
          <width>100</width>
          <height>100</height>
         </size>
        </property>
        <property name="autoFillBackground">
         <bool>true</bool>
        </property>
       </widget>
      </item>
      <item row="4" column="0">
       <layout class="QVBoxLayout" name="xyTranslation">
        <item>
         <widget class="QLabel" name="XTransDescriptionLabel">
          <property name="text">
           <string>Translation in X</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="xTransBox">
          <item>
           <widget class="QSlider" name="xTransSlider">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="xTransSlideLabel">
            <property name="text">
             <string>TextLabel</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="YTransDescriptionLabel">
          <property name="text">
           <string>Translation in Y</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="yTransBox">
          <item>
           <widget class="QSlider" name="yTransSlider">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="yTransSlideLabel">
            <property name="text">
             <string>TextLabel</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item row="5" column="0">
       <layout class="QGridLayout" name="gridLayout_2">
        <item row="0" column="2">
         <widget class="QLabel" name="label">
          <property name="text">
           <string>Red</string>
          </property>
         </widget>
        </item>
        <item row="3" column="2">
         <widget class="QLabel" name="label_3">
          <property name="text">
           <string>Blue</string>
          </property>
         </widget>
        </item>
        <item row="3" column="0">
         <widget class="QSlider" name="blueSlider">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="0">
         <widget class="QSlider" name="greenSlider">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="2" column="2">
         <widget class="QLabel" name="label_2">
          <property name="text">
           <string>Green</string>
          </property>
         </widget>
        </item>
        <item row="0" column="0">
         <widget class="QSlider" name="redSlider">
          <property name="maximum">
           <number>255</number>
          </property>
          <property name="orientation">
           <enum>Qt::Horizontal</enum>
          </property>
         </widget>
        </item>
        <item row="0" column="1">
         <widget class="QSpinBox" name="redSpinBox">
          <property name="maximum">
           <number>255</number>
          </property>
         </widget>
        </item>
        <item row="2" column="1">
         <widget class="QSpinBox" name="greenSpinBox">
          <property name="maximum">
           <number>255</number>
          </property>
         </widget>
        </item>
        <item row="3" column="1">
         <widget class="QSpinBox" name="blueSpinBox">
          <property name="maximum">
           <number>255</number>
          </property>
         </widget>
        </item>
       </layout>
      </item>
      <item row="1" column="2">
       <widget class="QPushButton" name="animalButton">
        <property name="text">
         <string>Spawn Animal</string>
        </property>
       </widget>
      </item>
      <item row="1" column="1">
       <widget class="QPushButton" name="animateButton">
        <property name="text">
         <string>Animate</string>
        </property>
       </widget>
      </item>
      <item row="4" column="2">
       <layout class="QVBoxLayout" name="xyScale">
        <item>
         <widget class="QLabel" name="xScaleDescriptionLabel">
          <property name="text">
           <string>Scale in X</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_8">
          <item>
           <widget class="QSlider" name="xScaleSlider">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="xScaleLabel">
            <property name="text">
             <string>TextLabel</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
        <item>
         <widget class="QLabel" name="yScaleDescriptionLabel">
          <property name="text">
           <string>Scale in Y</string>
          </property>
         </widget>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_9">
          <item>
           <widget class="QSlider" name="yScaleSlider">
            <property name="orientation">
             <enum>Qt::Horizontal</enum>
            </property>
           </widget>
          </item>
          <item>
           <widget class="QLabel" name="yScaleLabel">
            <property name="text">
             <string>TextLabel</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item row="1" column="0">
       <layout class="QVBoxLayout" name="Rotation">
        <property name="sizeConstraint">
         <enum>QLayout::SetFixedSize</enum>
        </property>
        <item>
         <widget class="QLabel" name="RotDescriptionLabel">
          <property name="text">
           <string>Rotation in Degrees</string>
          </property>
         </widget>
        </item>
        <item alignment="Qt::AlignLeft">
         <widget class="QDial" name="rotationDial"/>
        </item>
        <item>
         <layout class="QHBoxLayout" name="horizontalLayout_5">
          <item>
           <widget class="QLabel" name="rotationLabel">
            <property name="layoutDirection">
             <enum>Qt::LeftToRight</enum>
            </property>
            <property name="text">
             <string>TextLabel</string>
            </property>
           </widget>
          </item>
         </layout>
        </item>
       </layout>
      </item>
      <item row="2" column="1">
       <widget class="QSlider" name="timelineSlider">
        <property name="maximumSize">
         <size>
          <width>100</width>
          <height>16777215</height>
         </size>
        </property>
        <property name="maximum">
         <number>3</number>
        </property>
        <property name="orientation">
         <enum>Qt::Horizontal</enum>
        </property>
       </widget>
      </item>
     </layout>
    </widget>
   </item>
  </layout>
 </widget>
 <layoutdefault spacing="6" margin="11"/>
 <customwidgets>
  <customwidget>
   <class>MyGLWidget</class>
   <extends>QWidget</extends>
   <header location="global">myglwidget.h</header>
   <container>1</container>
   <slots>
    <slot>setXRotation(int)</slot>
    <slot>setZRotation(int)</slot>
    <slot>setYRotation(int)</slot>
   </slots>
  </customwidget>
 </customwidgets>
 <resources/>
 <connections>
  <connection>
   <sender>xScaleSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Window</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>940</x>
     <y>712</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yScaleSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Window</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>940</x>
     <y>766</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xTransSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Window</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>137</x>
     <y>712</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yTransSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>Window</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>137</x>
     <y>766</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>animalButton</sender>
   <signal>clicked()</signal>
   <receiver>Window</receiver>
   <slot>update()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>924</x>
     <y>591</y>
    </hint>
    <hint type="destinationlabel">
     <x>513</x>
     <y>399</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>redSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>86</x>
     <y>802</y>
    </hint>
    <hint type="destinationlabel">
     <x>414</x>
     <y>802</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>greenSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>greenSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>107</x>
     <y>835</y>
    </hint>
    <hint type="destinationlabel">
     <x>414</x>
     <y>835</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blueSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>blueSpinBox</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>94</x>
     <y>868</y>
    </hint>
    <hint type="destinationlabel">
     <x>414</x>
     <y>868</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>redSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>redSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>414</x>
     <y>802</y>
    </hint>
    <hint type="destinationlabel">
     <x>109</x>
     <y>802</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>greenSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>greenSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>414</x>
     <y>835</y>
    </hint>
    <hint type="destinationlabel">
     <x>111</x>
     <y>835</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>blueSpinBox</sender>
   <signal>valueChanged(int)</signal>
   <receiver>blueSlider</receiver>
   <slot>setValue(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>414</x>
     <y>868</y>
    </hint>
    <hint type="destinationlabel">
     <x>139</x>
     <y>868</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>animateButton</sender>
   <signal>clicked()</signal>
   <receiver>animateButton</receiver>
   <slot>click()</slot>
   <hints>
    <hint type="sourcelabel">
     <x>563</x>
     <y>592</y>
    </hint>
    <hint type="destinationlabel">
     <x>521</x>
     <y>581</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>rotationDial</sender>
   <signal>valueChanged(int)</signal>
   <receiver>rotationLabel</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>123</x>
     <y>603</y>
    </hint>
    <hint type="destinationlabel">
     <x>210</x>
     <y>627</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xTransSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>xTransSlideLabel</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>386</x>
     <y>712</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>712</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yTransSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>yTransSlideLabel</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>371</x>
     <y>766</y>
    </hint>
    <hint type="destinationlabel">
     <x>454</x>
     <y>766</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>xScaleSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>xScaleLabel</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>940</x>
     <y>712</y>
    </hint>
    <hint type="destinationlabel">
     <x>1002</x>
     <y>712</y>
    </hint>
   </hints>
  </connection>
  <connection>
   <sender>yScaleSlider</sender>
   <signal>valueChanged(int)</signal>
   <receiver>yScaleLabel</receiver>
   <slot>setNum(int)</slot>
   <hints>
    <hint type="sourcelabel">
     <x>940</x>
     <y>766</y>
    </hint>
    <hint type="destinationlabel">
     <x>1002</x>
     <y>766</y>
    </hint>
   </hints>
  </connection>
 </connections>
</ui>
